File: sdkwork-ai-sdk-typescript/src/api/context.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { ChatWithContextRequest, ContextCreateRequest, ContextResponse } from '../types';


export class ContextApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Chat with context */
  async chatWith(context_id: string | number, body: ChatWithContextRequest): Promise<ContextResponse> {
    return this.client.post<ContextResponse>(aiApiPath(`/context/${context_id}/chat/completions`), body);
  }

/** Create context */
  async createContext(body: ContextCreateRequest): Promise<ContextResponse> {
    return this.client.post<ContextResponse>(aiApiPath(`/context/create`), body);
  }

/** Get context */
  async getContext(context_id: string | number): Promise<ContextResponse> {
    return this.client.get<ContextResponse>(aiApiPath(`/context/${context_id}`));
  }

/** Delete context */
  async deleteContext(context_id: string | number): Promise<void> {
    return this.client.delete<void>(aiApiPath(`/context/${context_id}`));
  }
}

export function createContextApi(client: HttpClient): ContextApi {
  return new ContextApi(client);
}

File: sdkwork-ai-sdk-go/api/context.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type ContextApi struct {
    client *sdkhttp.Client
}

func NewContextApi(client *sdkhttp.Client) *ContextApi {
    return &ContextApi{client: client}
}

// Chat with context
func (a *ContextApi) ChatWith(contextId string, body sdktypes.ChatWithContextRequest) (sdktypes.ContextResponse, error) {
    raw, err := a.client.Post(AiApiPath(fmt.Sprintf("/context/%s/chat/completions", contextId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.ContextResponse
        return zero, err
    }
    return decodeResult[sdktypes.ContextResponse](raw)
}

// Create context
func (a *ContextApi) CreateContext(body sdktypes.ContextCreateRequest) (sdktypes.ContextResponse, error) {
    raw, err := a.client.Post(AiApiPath("/context/create"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.ContextResponse
        return zero, err
    }
    return decodeResult[sdktypes.ContextResponse](raw)
}

// Get context
func (a *ContextApi) GetContext(contextId string) (sdktypes.ContextResponse, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/context/%s", contextId)), nil, nil)
    if err != nil {
        var zero sdktypes.ContextResponse
        return zero, err
    }
    return decodeResult[sdktypes.ContextResponse](raw)
}

// Delete context
func (a *ContextApi) DeleteContext(contextId string) (struct{}, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/context/%s", contextId)), nil, nil)
    if err != nil {
        var zero struct{}
        return zero, err
    }
    return decodeResult[struct{}](raw)
}
